vmicore/src/lib/plugins/PluginSystem.cpp

```diff
@@ -121,10 +121,13 @@ namespace VmiCore
         registeredShutdownCallbacks.push_back(shutdownCallback);
     }
 
-    std::shared_ptr<IBreakpoint> PluginSystem::createBreakpoint(
-        uint64_t targetVA, uint64_t processDtb, const std::function<BpResponse(IInterruptEvent&)>& callbackFunction)
+    std::shared_ptr<IBreakpoint>
+    PluginSystem::createBreakpoint(uint64_t targetVA,
+                                   uint64_t processDtb,
+                                   const std::function<BpResponse(IInterruptEvent&)>& callbackFunction,
+                                   bool processDependent)
```

Comment:
nit: Imho the name `processDependent` raises questions. I would flip the bool and simply call it `global` or something like that. I would also reorder the signature so the callback is the last param. It makes more sense, especially if you decided to provide an inline lambda here.

Reply:
Come to think of it: I don't see why the user would want to register global breakpoints. I think we should not change the API here.

Reply:
I think it is a nice feature. For example you want to monitor a function call from a malware component which we do not trace (e.g. currently it could be because of process injection, later on it could be because the component is started by a system service).

Reply:
For now the global flag is kept in our internals to increase performance but is no longer exposed to the plugins. They create per default process dependent breakpoints.